LFS-939: Add the ability to specify "compact" display for questions with predefined options - all options on one line

marta- opened the pull request:
To test, use the questionnaire wizard to set the `compact` property to `true` for any question that has `dataType` : `boolean` or `displayType`: `list` or `list+input`, then create a form for that questionnaire to see how the options are displayed.

marta- committed:
LFS-939: Add the ability to specify "compact" display for questions with predefined options - all options on one line

marta- committed:
LFS-939: Add the ability to specify "compact" display for questions with predefined options - all options on one line
On narrow screens, display options one per line regardless of the `compact` setting.

sashaandjic reviewed (approved):
tested

sdumitriu commented on modules/data-entry/src/main/resources/SLING-INF/nodetypes/dataentry.cnd:
```diff
@@ -182,6 +182,9 @@
   - minValue (double)
   - maxValue (double)
 
+  // For questions that have a list of options, whether the options are displayed on a single line (compact) or one per line (default)
```
```suggestion
  // For questions that have a list of options, whether all the options are displayed on a single line (compact), or one per line (default)
```

Still, this isn't entirely true, since they may not all fit on one line.

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/QuestionnaireStyle.jsx:
```diff
@@ -188,6 +188,30 @@ const questionnaireStyle = theme => ({
+          flexDirection: "row",
+        }
+      },
+      "& .MuiListItem-root" : {
+        [theme.breakpoints.up('sm')]: {
+          display: "inline-flex",
+          width: "auto",
+        },
+      },
+      "& .MuiList-root" : {
+        [theme.breakpoints.up('sm')]: {
+          display: "inline-block",
+        },
+      },
```
I would reorder these in order of nesting (List, ListItem). Or even better, nest the style as well (ListItem inside List inside FormGroup).

sdumitriu reviewed (commented):
I think the spacing between "label checkbox label" is a bit too confusing. I would like to have a pretty border or shading around an option when hovering it, to make it more obvious which checkbox belongs to each label.

marta- committed:
LFS-939: "compact" display for questions

Updated documentation/comments according to review

Co-authored-by: Sergiu Dumitriu <sdumitriu@users.noreply.github.com>

marta- committed:
LFS-939: Add the ability to specify "compact" display for questions with predefined options - all options on one line
Adressing review comments - reordered style rules

marta- commented:
> I think the spacing between "label checkbox label" is a bit too confusing. I would like to have a pretty border or shading around an option when hovering it, to make it more obvious which checkbox belongs to each label.

Spacing fixes and hover shading were going to be part of 955, but I included some of them here in commit 32f1aa6.

marta- committed:
LFS-939: Add the ability to specify "compact" display for questions with predefined options - all options on one line
Improved markup and simplified styling

marta- commented:
> I think the spacing between "label checkbox label" is a bit too confusing. I would like to have a pretty border or shading around an option when hovering it, to make it more obvious which checkbox belongs to each label.

Spacing improvements and hover shading were going to be part of 955. I find the spacing pretty clear especially for the questions where it actually makes sense to use compact layout.

sashaandjic commented:
Only one minor thing. The free text line went a bit down in the last iteration.
![image](https://user-images.githubusercontent.com/23243244/109543892-e58e7880-7a94-11eb-8d2c-6d686d225c36.png)

marta- committed:
LFS-939: Add the ability to specify "compact" display for questions with predefined options - all options on one line
Addressing comments - Bringing in more changes from LFS955 to fix alignment issue introduced in previous commit

sashaandjic reviewed (approved):
tested

acrowthe reviewed (approved):
Looks good
